core/consensus/src/message.rs

```diff
@@ -211,6 +213,55 @@ impl<R: Rpc + 'static, S: Storage + 'static> MessageHandler for PullBlockRpcHand
     }
 }
 
+#[derive(Debug)]
+pub struct PullProofRpcHandler<R, S> {
+    rpc:     Arc<R>,
+    storage: Arc<S>,
+}
+
+impl<R, S> PullProofRpcHandler<R, S>
+where
+    R: Rpc + 'static,
+    S: Storage + 'static,
+{
+    pub fn new(rpc: Arc<R>, storage: Arc<S>) -> Self {
+        PullProofRpcHandler { rpc, storage }
+    }
+}
+
+#[async_trait]
+impl<R: Rpc + 'static, S: Storage + 'static> MessageHandler for PullProofRpcHandler<R, S> {
+    type Message = FixedHeight;
+
+    async fn process(&self, ctx: Context, height: FixedHeight) {
+        let height = height.inner;
+        let latest_proof = self.storage.get_latest_proof().await;
```

Comment:
```
current_h := self. storage.get_latest_height()

if current_h == height {
      self.storage.get_latest_proof()
}
else if current_h > height {
        self.storage.get_block_by_height(h).proof
} else {
  return GetNoneErr
}
```

Reply:
Cannot use ? operator to throw in MessageHandler.  Have to catch them all